inspirehep/modules/records/utils.py

```diff
@@ -414,6 +420,25 @@ def populate_authors_full_name_unicode_normalized(json):
         })
 
 
+def populate_facet_author_name(json):
+    """Populate the ``facet_author_name`` field of Literature records."""
+    linked_authors = list(get_linked_records_in_field(json, 'authors.record'))
+    unlinked_authors = [author for author in json.get('authors', []) if 'record' not in author]
+    result = []
+
+    for author in linked_authors:
+        bai = [id['value'] for id in author['ids'] if id['schema'] == 'INSPIRE BAI'][0]
+        if 'preferred_name' in author['name']:
```

Comment:
we could use https://github.com/inspirehep/inspire-utils/blob/e055a50e6f848643b38b68fe5f7f4ad913cb0767/inspire_utils/record.py#L33-L86

```suggest
author_preffered_name = get_value(author, 'name. preferred_name')
if author_preffered_name:
    result.append(u'{}_{}'.format(bai, author_preffered_name))
```